tests/network/primary_pod_network.go

```diff
@@ -70,17 +70,62 @@ var _ = SIGDescribe("[Serial]Primary Pod Network", func() {
 		})
 
 		Context("VMI connected to the pod network using bridge binding", func() {
-			var vmi *v1.VirtualMachineInstance
+			When("Guest Agent exists", func() {
+				var (
+					vmi   *v1.VirtualMachineInstance
+					vmiIP = func() (string, error) {
+						vmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+						if err != nil {
+							return "", err
+						}
+						return vmi.Status.Interfaces[0].IP, nil
+					}
 
-			BeforeEach(func() {
-				vmi = setupVMI(virtClient, vmiWithBridgeBinding())
-			})
+					vmiIPs = func() ([]string, error) {
+						vmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+						if err != nil {
+							return nil, err
+						}
+						return vmi.Status.Interfaces[0].IPs, nil
+					}
+				)
+				BeforeEach(func() {
+					var err error
+					vmi, err = newFedoraWithGuestAgentAndBridgeBinding()
+					Expect(err).NotTo(HaveOccurred())
 
-			AfterEach(func() {
-				cleanupVMI(virtClient, vmi)
+					vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+					Expect(err).NotTo(HaveOccurred())
+					tests.WaitForSuccessfulVMIStart(vmi)
+					tests.WaitAgentConnected(virtClient, vmi)
+				})
+
+				AfterEach(func() {
+					cleanupVMI(virtClient, vmi)
+				})
+
+				It("should report PodIP/s as its own on interface status", func() {
+					vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+					Eventually(vmiIP).Should(WithTransform(removeCIDR, Equal(vmiPod.Status.PodIP)), "should contain VMI Status IP as Pod status ip")
+					Eventually(vmiIPs).ShouldNot(BeEmpty(), "should have some IPs at status")
+					ipFromVMI, err := vmiIP()
+					Expect(err).ToNot(HaveOccurred())
```

Comment:
I liked the neat usage here, but:
- We call the API too many times. For each entry (IP, IPs) and multiple times (line 111).
- Why do you need to check for vmiIPs() emptiness? Even if it is empty, the `ContainElements` should take care of that, no?

Reply:
Done